src/run.js

```diff
@@ -485,15 +485,29 @@ const minimalcss = async options => {
     );
   }
 
+  // This is the protection against ever looking up the same CSS selector
+  // more than once. We can confidently pre-populate it with a couple that
+  // we're confident about.
+  const decisionsCache = { '*': true, body: true, html: true, '': true };
+
   // Now, let's loop over ALL links and process their ASTs compared to
   // the DOMs.
-  const decisionsCache = {};
   const isSelectorMatchToAnyElement = selectorString => {
     // Here's the crucial part. Decide whether to keep the selector
     // Find at least 1 DOM that contains an object that matches
     // this selector string.
     return doms.some(dom => {
       try {
+        // console.log(
+        //   'SELECTOR:',
+        //   selectorString,
+        //   selectorString in decisionsCache
+        // );
+        if (selectorString in decisionsCache) {
+          throw new Error(
```

Comment:
if you throw Error it will gather stacktrace along the way, if you want to use throw for control flow you may throw plain string (theoretically it is cheaper). But I would simply `return true` in this case instead of throw

Reply:
I just removed it now. It was there to "check myself". In an earlier draft of the code, something wasn't right and lookups were made when they had already been made.